Source/WebCore/page/Page.cpp

```diff
@@ -2015,6 +2024,9 @@ void Page::renderingUpdateCompleted()
         scheduleRenderingUpdateInternal();
         m_unfulfilledRequestedSteps = { };
     }
+
+    if (!isUtilityPage())
+        m_opportunisticTaskScheduler->reschedule(m_lastRenderingUpdateTimestamp + Seconds(1. / m_displayNominalFramesPerSecond.value_or(60)));
```

Comment:
We always assume 60fps by default?
Based on FIXME in WebPageProxy::swapToProvisionalPage, macOS will always use this default value?

Reply:
Oh, that's a good point — I think that 60 is effectively the default value, since this is plumbed down from the UI process, which otherwise falls back to `FullSpeedFramesPerSecond` (60 fps) in the case where the display link is null for any reason:

```
FramesPerSecond DisplayLink::nominalFramesPerSecondFromDisplayLink(CVDisplayLinkRef displayLink)
{
    CVTime refreshPeriod = CVDisplayLinkGetNominalOutputVideoRefreshPeriod(displayLink);
    if (!refreshPeriod.timeValue)
        return FullSpeedFramesPerSecond;

    FramesPerSecond result = round((double)refreshPeriod.timeScale / (double)refreshPeriod.timeValue);
    return result ?: FullSpeedFramesPerSecond;
}
```

In practice, I don't think there's really a valid scenario where a (non-utility) `Page` would have a null `m_displayNominalFramesPerSecond` — we can probably just gate the rescheduling on having a non-null `m_displayNominalFramesPerSecond`:

```
void Page::renderingUpdateCompleted()
{
…
    if (!isUtilityPage() && m_displayNominalFramesPerSecond)
        m_opportunisticTaskScheduler->reschedule(m_lastRenderingUpdateTimestamp + Seconds(1. / *m_displayNominalFramesPerSecond));
}
```